nncf/compression_method_api.py

```diff
@@ -329,6 +329,12 @@ def _nncf_module_types_to_compress(self) -> List[str]:
                 filtered_nncf_module_names_list.append(module.__name__)
         return filtered_nncf_module_names_list
 
+    def are_frozen_layers_allowed(self) -> bool:
+        return False
+
+    def algo_name(self) -> str:
```

Comment:
Shouldn't extend API just for that. Find a way to reuse the string values that are given as `@COMPRESSION_ALGORITHM.register()` parameters.

Reply:
First, this decorator has nothing to do with the actual Builder class. It just put the name on the registry.
Second, the name can be not obvious, some logic may require -
https://github.com/openvinotoolkit/nncf/pull/382/files#diff-c2d882888dac8c1721e41bf7d650e6e3fc137181f13f835c25c3e0b27809eefeR1101

Reply:
The user won't interact with this function in any feasible way. This is a builder function, after all, and what user finally gets are controllers. Remove this function, and instead extend the `are_frozen_layers_allowed` function return value with a string specifying the reason why these are not allowed. After an invocation of `are_frozen_layers_allowed` (which will be done somewhere inside the builder code anyway), print the reason as a warning message.